airbyte-webapp/src/core/jsonSchema/schemaToYup.ts

```diff
@@ -34,20 +34,22 @@ export const buildYupFormForJsonSchema = (
     | null = null;
 
   if (jsonSchema.oneOf && uiConfig && propertyPath) {
-    const selectedSchema =
-      jsonSchema.oneOf.find((condition) => {
-        if (typeof condition !== "boolean") {
-          return uiConfig[propertyPath]?.selectedItem === condition.title;
-        }
-        return false;
-      }) ?? jsonSchema.oneOf[0];
+    let selectedSchema = jsonSchema.oneOf.find(
+      (condition) =>
+        typeof condition !== "boolean" &&
+        uiConfig[propertyPath]?.selectedItem === condition.title
+    );
+
+    // Select first oneOf path if no item selected
+    selectedSchema = selectedSchema ?? jsonSchema.oneOf[0];
+
     if (selectedSchema && typeof selectedSchema !== "boolean") {
       return buildYupFormForJsonSchema(
         { type: jsonSchema.type, ...selectedSchema },
         uiConfig,
         jsonSchema,
         propertyKey,
-        propertyPath ? `${propertyPath}.${propertyKey}` : propertyKey
```

Comment:
We're having the same logic two times more further down. Once inside an object and once inside the array branch. For the object the `propertyKey` we're actually having is not the one passed into this method, but for array it's the same as passed into, and thus the logic the same as here. Could you please explain (and give some example schemas to illustrate it), why the logic needs adjustment in this place, but for the other two calls to `buildYupFormForJsonSchema` further down it's the correct behavior?

Reply:
@timroes let me know if it's still unclear. I will try to provide a more accurate answer